### DISCUSSION TITLE
[Enhancement] [table] el-table获取当前滚动条所在的位置

### DISCUSSION THREAD

**Discussion by arkfancy** [element-plus/element-plus#9679] [Feature Request]
Feature type: **`Enhancement`**

## Details

- Enhanced component name: `table`

## Description

目前table只有scrollTo相关的方法可以设置滚动条滚动到哪个位置，没办法反向获取当前滚动条所在的位置。
希望增加类似@scroll-change这样的事件，当鼠标滚动了表格内的滚动条时，能获取当前滚动所在的位置

### Additional comments

_(empty)_

**Comment by arkfancy** [comment]
![image](https://user-images.githubusercontent.com/30465365/188786494-71514ca8-80b8-4d27-874e-d828bb40b9da.png)
目前看tableRef里面有scrollBarRef，但是里面只有el-scroll的相关方法，没有相关事件，将事件抛出来应该就可以实现监听了

现在的实现方式是这样
tableRef.value.scrollBarRef.wrap$.onscroll = (event) => {
  console.log(event.target.scrollLeft)
}

**Reply by wrdickson** [reply]
This is very helpful.  

this.$refs.scrollBarRef.wrap$.onscroll = (event) => {
     left = event.target.scrollLeft
     top= event.target.scrollTop
 }
Is used to store the table scroll position in Pinia or whatever.
Then, when instantiating the component again

this.$refs.scrollBarRef.setScrollLeft( left )
this.$refs.scrollBarRef.setScrollTop( top )

should replicate the scroll state of the table.

I was not aware of "wrap$.onscroll" and your code helped me considerably.  Thank you!

**Reply by arkfancy** [reply]
in 2.2.22 
it was replaced by 'wrapRef'
![image](https://user-images.githubusercontent.com/30465365/202974286-3e061073-1e17-41e3-93c8-fe437cb4d7bf.png)

**Reply by wrdickson** [reply]
I just updated to 2.2.22 and am using "wrapRef" not "wrap$".  It works perfectly.

Thank you, again!  I do not see any of this in the English documentation, so I am grateful for your work and for this update.
